File: SchoolManagementApi/DTOs/Subjects/CreateSubjectDto.cs
namespace SchoolManagementApi.DTOs.Subjects;

public class CreateSubjectDto
{
    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    public int TeacherId { get; set; }
    public int Credits { get; set; } = 3;
}

File: SchoolManagementApiNode/src/subjects/dto/create-subject.dto.ts
import { IsNotEmpty, IsOptional, Min, MaxLength } from 'class-validator';
import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';

export class CreateSubjectDto {
  @ApiProperty({ example: 'Matemáticas', maxLength: 100 })
  @IsNotEmpty()
  @MaxLength(100)
  name: string;

  @ApiPropertyOptional({ example: 'Curso de matemáticas básicas' })
  @IsOptional()
  description?: string;

  @ApiProperty({ example: 1 })
  @IsNotEmpty()
  teacherId: number;

  @ApiProperty({ example: 3 })
  @Min(1)
  credits: number = 3;
}
